### DISCUSSION TITLE
Unexpected new line after hyphen

### DISCUSSION THREAD

**Discussion by me-oct** [thinreports/thinreports#20] [Q&A]
Thank you for providing such a powerful and helpful PDF generator.

We run a system with the thinreports 0.10.2 to create PDF files.
We have encountered a same problem with the issue "Unexpected new line after hyphen" discussed here; https://github.com/thinreports/thinreports-generator/issues/50.
I understand the cause, but there is no solution found in the discussion.
What we can do as a workaround at this moment is to replace hyphens with other visually similar characters.
I would appreciate if you could give me a solution to the problem or any better workaround.

**Comment by hidakatsuya** [comment]
Thinreports uses Prawn to generate PDFs, but there is no option in Prawn to disable this behavior (hyphenation).

Unfortunately, I think the only current workaround is to replace the hyphen with a visually similar character such as a non-breaking hyphen.

```ruby
report.item(:text).value(text.gsub("\u002d", "\u2011"))
```

**Reply by me-oct** [reply]
hidakatsuya-san, 

Thank you for the clear explanation.
OK. We will take the workaround option.

**Comment by hidakatsuya** [comment]
If you like, you can try [the prawn-disable_word_break gem](https://github.com/hidakatsuya/prawn-disable_word_break) for an alternative solution.

```diff
# Gemfile
gem 'thinreports'
+ gem 'prawn-disable_word_break'
```

**Reply by me-oct** [reply]
It's a newly build gem and what we wanted. Thank you for the useful gem.

It's a pity that we use ruby 2.4 and thinreports 0.10.2 + prawn 2.2.2 for the system, which are not supported by the gem.
We will try it at the next opportunity.
